target/linux/armsr/base-files/etc/inittab

```diff
@@ -1,7 +1,7 @@
 ::sysinit:/etc/init.d/rcS S boot
 ::shutdown:/etc/init.d/rcS K shutdown
 ttyAMA0::askfirst:/usr/libexec/login.sh
-ttyS0::askfirst:/usr/libexec/login.sh
+@GRUB_SERIAL@::askfirst:/usr/libexec/login.sh
```

Comment:
https://github.com/openwrt/openwrt/blob/main/target/linux/armsr/image/Makefile#L20 and https://github.com/openwrt/openwrt/blob/main/target/linux/x86/image/Makefile#L21 currently assume `GRUB_SERIAL` can be empty, which implies this is a valid configuration.

Maybe instead of erroring out append/replace the whole line instead when `GRUB_SERIAL` is empty?

Reply:
I think it is always set on the x86 and armsr target. I think we should unify this handling.

Reply:
Resolved by making the above two files unconditional.